pkg/releaseutil/manifest.go

```diff
@@ -45,20 +43,115 @@ func SplitManifests(bigFile string) map[string]string {
 	res := map[string]string{}
 	// Making sure that any extra whitespace in YAML stream doesn't interfere in splitting documents correctly.
 	bigFileTmp := strings.TrimSpace(bigFile)
-	docs := sep.Split(bigFileTmp, -1)
+	docs := splitDocs(bigFileTmp)
 	var count int
 	for _, d := range docs {
-		if d == "" {
-			continue
-		}
-
-		d = strings.TrimSpace(d)
 		res[fmt.Sprintf(tpl, count)] = d
 		count = count + 1
 	}
 	return res
 }
 
+func splitDocs(bigFile string) []string {
+	docs := make([]string, 0)
+	docStartIdx := 0
+	docEndIdx := 0
+	var char rune
+	var width int
+	// using a 'raw' for loop to avoid creating a complicated statemachine
+	i := 0
+	//if strings.HasPrefix(bigFile, "---") {
+	//	i = 3
+	//}
+
+	for i < len(bigFile) {
```

Comment:
If I was writing this, I'd be tempted to look at creating a scanner and/or state machine to make this more readable. I think something could be done that wouldn't hurt performance.

Reply:
Today i tried two implementations, one with a 'scanner/iterator' but this was clearer to read.

But then looking at the profiler output i saw that 'strings.Prefix' is highly optimized so i tried a 'naive' implementation which was actually twice as fast.

I.e. we now check at very subslice that it start with "\n---".

Reply:
It is a lot easier to read now, thanks!